Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,230 @@
+        /// <param name="filename">Name of the file being processed</param>
+        /// <param name="stringEntries">List of string entries to be populated by the method</param>
+        /// <param name="entryBuilder">String builder that contains the current entry</param>
+        public static void ProcessBuffer(char[] buffer, PxFileSyntaxConf syntaxConf, LineCharacterState state, string filename, List<StringValidationEntry> stringEntries, StringBuilder entryBuilder)
+        {
+            foreach (char currentCharacter in buffer)
+            {
+                if (IsEndOfMetadataSection(currentCharacter, syntaxConf, entryBuilder))
+                {
+                    break;
+                }
+                UpdateLineAndCharacter(currentCharacter, syntaxConf, state);
+                CheckForEntryEnd(currentCharacter, syntaxConf, state, filename, stringEntries, entryBuilder);
+            }
+        }
+
+        private static List<StringValidationEntry> BuildStringEntries(Stream stream, Encoding encoding, PxFileSyntaxConf syntaxConf, string filename, int bufferSize)
+        {
+            SyntaxValidation.LineCharacterState state = new() { Line = 0, Character = 0 };
+            stream.Seek(0, SeekOrigin.Begin);
+            using StreamReader reader = new(stream, encoding);
+            List<StringValidationEntry> stringEntries = [];
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+
+            while (reader.Read(buffer, 0, bufferSize) > 0)
+            {
+                ProcessBuffer(buffer, syntaxConf, state, filename, stringEntries, entryBuilder);
+                Array.Clear(buffer, 0, buffer.Length);
+            }
+            return stringEntries;
+        }
+
+        private static bool IsEndOfMetadataSection(char currentCharacter, PxFileSyntaxConf syntaxConf, StringBuilder entryBuilder)
+        {
+            if (currentCharacter == syntaxConf.Symbols.KeywordSeparator)
+            {
+                string stringEntry = entryBuilder.ToString();
+                // When DATA keyword is reached, metadata parsing is complete
+                return SyntaxValidationUtilityMethods.CleanString(stringEntry).Equals(syntaxConf.Tokens.KeyWords.Data);
+            }
+            return false;
+        }
+
+        private static void UpdateLineAndCharacter(char currentCharacter, PxFileSyntaxConf syntaxConf, LineCharacterState state)
+        {
+            if (currentCharacter == syntaxConf.Symbols.LineBreak)
+            {
+                state.Line++;
+                state.Character = 0;
+            }
+            else
+            {
+                state.Character++;
+            }
+        }
+
+        private static void CheckForEntryEnd(char currentCharacter, PxFileSyntaxConf syntaxConf, LineCharacterState state, string filename, List<StringValidationEntry> stringEntries, StringBuilder entryBuilder)
```

Comment:
This function should be renamed to match the actual behaviour. It would also be better if the entry is returned from the function, mutating the parameters is harder to undertand and more prone to unvanted side effects.